rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,302 @@
+    },
+    "departureTime": "25:45:00"
+  }
+}
+```
+
+Note that the date of the the event is 2023-01-23, but the service date is 2023-01-22. The hour in the time itself is greater than 24 hours, to reflect that the time is the next calendar day.
+### Splitting a 2-car train into two 1-car trips
+- Inspector Alice (badge number: 123) is working at Boston College
+- There are two scheduled trips: 9:55am and 10:00am
+- Operator A (badge number: 456) and Operator B (badge number: 567) are scheduled to perform the 9:55am
+- Only 1 trainset (3800 and 3850) is available
+- In order to maintain headways, Inspector Alice will split the trainset
+- Operator A will take 3800 and run the 9:55am
+- Operator B will take 3850 and run the 10:00am
+
+```json
+{
+  "specversion": "1.0",
+  "type": "com.mbta.ctd.glides.operator-signed-in.v1",
+  "source": "glides.mbta.com",
+  "id": "19fdb184-7dd6-4664-8472-04bd6177ec44",
+  "time": "2023-01-20T09:45:10-05:00",
+  "data": {
+    "actor": {
+      "username": "ainspector",
+      "badgeNumber": "123"
+    },
+    "operator": {
+      "badgeNumber": "456"
+    },
+    "signInAt": "2023-01-20T09:45:00-05:00"
+  }
+}
+# further specversion, source, time, and ID fields elided from the top level
+# actor elided from underneath "data"
+{
+  "type": "com.mbta.ctd.glides.operator-signed-in.v1",
+  "data": {
+    "operator": {
+      "badgeNumber": "567"
+    },
+    "signInAt": "2023-01-20T09:45:00-05:00"
+  }
+}
+{
+  "type": "com.mbta.ctd.glides.trip-dropped.v1",
+  "data": {
+    "tripKey": {
+      "serviceDate": "2022-01-20",
+      "stop": "place-lake",
+      "scheduledTime": "10:00:00"
+    },
+    "reason": "ran as single"
+  }
+}
+{
+  "type": "com.mbta.ctd.glides.trip-updated.v1",
+  "data": {
+    "tripKey": {
+      "serviceDate": "2022-01-20",
+      "stop": "place-lake",
+      "scheduledTime": "09:55:00"
+    },
+    "operators": [
+      {"badgeNumber": "456"}
+    ],
+    "consist": [
+      {"carNumber": "3800"}
+    ]
+  }
+}
+{
+  "type": "com.mbta.ctd.glides.trip-added.v1",
+  "time": "2023-01-20T09:50:15-05:00",
+  "data": {
+    "actor": {
+      "username": "ainspector",
+      "badgeNumber": "123"
+    },
+    "tripKey": {
+      "service-date": "2022-01-20",
+      "id": "ADDED-1"
+    },
+    "stop": "place-lake",
+    "time": "10:00:00",
+    "tripType": "round",
+    "consist": [
+      {"carNumber": "3850"}
+    ],
+    "operators": [
+      {"badgeNumber": "567"}
+    ]
+  }
+}
+```
+
+In this instance, the inspector dropped the 10:00am trip, and created a new trip with the same departure time. Ideally, RTR will treat this the same as if the 10:00am were updated to be a 1-car trip, but inspectors do not always enter the data in that fashion.
+
+# Drawbacks
+- This requires Glides to include new functionality to write their messages into a Kinesis stream. However, they already use `ex_aws` so additional messages should not be hard to add.
+ 
+# Rationale and alternatives
```

Comment:
This should list the rationale for including operator information (both sign-in and assignment). I'm not sure what that rationale is, and I might disagree with it.

Reply:
+1. Including operator badges and sign ins adds a significant amount of volume to this RFC. We should have a consumer or reason in mind before committing to including it. The spec has to be forwards-compatible anyway, so it has to be possible to add operator information back in later if needed.

Reply:
From a discussion with JesseB, operators drive differently (some are faster/slower, for example) which could be of use for predictions. Including sign in is for Lamp, so that all of the data generated by inspectors flows into that system.

Reply:
It didn't seem like much complexity to me, as it's all information that Glides already has for both the inspector and the operator signing in. Can you say more about what's complex about this implementation?